nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -386,6 +392,38 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
     plan.poses.push_back(pose);
   }
 
+  // plot footprint path planned
+  if (_debug_visualizations) {
+    auto marker_array = std::make_unique<visualization_msgs::msg::MarkerArray>();
+    visualization_msgs::msg::Marker m = createMarker(_global_frame, _clock->now());
+
+    for (size_t i = 0; i < plan.poses.size(); i++) {
+      const double & x = plan.poses[i].pose.position.x;
+      const double & y = plan.poses[i].pose.position.y;
+      const double & yaw = tf2::getYaw(plan.poses[i].pose.orientation);
+
+      std::vector<geometry_msgs::msg::Point> footprint =
+        transformFootprintToEdges(x, y, yaw, _costmap_ros->getRobotFootprint());
```

Comment:
Pass this function the `plan.poses[i].pose` and do the 

```

      const double & x = plan.poses[i].pose.position.x;
      const double & y = plan.poses[i].pose.position.y;
      const double & yaw = tf2::getYaw(plan.poses[i].pose.orientation);
```

In the function itself. Since x, y, yaw are never used outside of that function, its scope is the function.